src/Circle.js

```diff
@@ -0,0 +1,95 @@
+import React, { useRef, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const DIRECTIONS = {
+    CLOCKWISE: 'clockwise',
+    ANTI_CLOCKWISE: 'antiClockwise',
+    BOTH_SIDES: 'bothSides',
+};
+
+const getSvgClassName = (strokePercentage, direction, className) => classNames(
+    'react-circle_svg',
+    `react-circle_${direction}`,
+    {
+        'react-circle_noStroke': strokePercentage === 0,
+    },
+    className,
+);
+
+const Circle = ({ className, direction, strokeWidth, strokePercentage, onTransitionEnd }) => {
+    const svgRef = useRef(null);
+    const [circleState, setCircleState] = useState({});
+    const [svgState, setSvgState] = useState({ className: getSvgClassName(strokePercentage, direction, className) });
+
+    useEffect(() => {
+        const { clientWidth: svgSize } = svgRef.current;
+
+        const center = svgSize / 2;
+        const radius = center - parseInt(strokeWidth, 10);
```

Comment:
If I remember correctly, there were times were radius would be negative and an error would show on the console.

Imagine we have a `strokeWidth` of 5 and svgSize is 0. Radius would be `-5`. We should keep the `Math.max` here.

Reply:
Yup 👍